### PR TITLE
[LC-375] Update prep list via icon-service

### PR DESCRIPTION
**Pull request by yakkle**
- get prep list via icon-service
 - do _load_preps_from_iiss when add block.
 - Reset leader if new leader doesn't equal current epoch leader
 - update Makefile for test
 - Fix the bug, which can't make up a block with just one node.
 - get peer ids hash before reset its subscribe connection.
 - delete response_temp from load peer from iiss
 - apply peer_manager list when getting node_type_by_peer_list
 - separate role switch checking method
 - delete filtering channel infos (from file) by peer id
 - clean up temporary notice logs

related PR : https://github.com/icon-project/loopchain/pull/211

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,9 +37,9 @@ develop: install generate-proto
 
 # pip install packages
 install:
-	$(PIP_INSTALL) git+https://github.com/icon-project/icon-service.git@master
+	$(PIP_INSTALL) git+https://github.com/icon-project/icon-service.git@IS-492-prep-testnet-support
 	$(PIP_INSTALL) git+https://github.com/icon-project/icon-commons.git@master
-	$(PIP_INSTALL) git+https://github.com/icon-project/icon-rpc-server.git@master
+	$(PIP_INSTALL) git+https://github.com/icon-project/icon-rpc-server.git@release-1.3
```
Comment (yakkle) [inline Makefile]: FIXME!

File: loopchain/channel/channel_service.py [hunk 2]
```diff
@@ -434,7 +432,44 @@ def __init_node_subscriber(self):
...
+
+        if self.__peer_manager.get_peer_ids_hash(peer_ids) == self.__peer_manager.peer_ids_hash():
+            utils.logger.debug(f"There is no change in peers.")
+            return
+
+        utils.logger.debug(f"Peer manager have to update with new list.")
+        self.__peer_manager.reset_peers(check_status=False)
+
+        order = 1
+        for rep_info in response["result"]["preps"]:
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: It doesn't remove peers not used anymore from peer manager and unsubscribe from broadcast scheduler.

File: loopchain/baseservice/peer_manager.py [hunk 3]
```diff
@@ -126,6 +126,34 @@ def __init__(self, channel_name):
         self.__leader_complain_count = 0
         self.__highest_block_height = -1    # for RS heartbeat
 
+    def peer_ids_hash(self):
+        """ It's temporary develop for Prep test net. This value will replace with Prep root hash.
+        peer_list =
```
Comment (yakkle) [inline loopchain/baseservice/peer_manager.py]: something strange... what is peer_list?

File: loopchain/baseservice/peer_manager.py [hunk 4]
```diff
@@ -126,6 +126,34 @@ def __init__(self, channel_name):
         self.__leader_complain_count = 0
         self.__highest_block_height = -1    # for RS heartbeat
 
+    def peer_ids_hash(self):
+        """ It's temporary develop for Prep test net. This value will replace with Prep root hash.
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]:
```python
from functools import reduce
from operator import add

peer_orders = self.peer_order_list[group_id]
peer_ids = (peer_orders[peer_order] for peer_order in sorted(peer_orders.keys()))
peer_ids_appended = reduce(add, peer_ids)
return self.get_peer_ids_hash(peer_ids_appended)
```

File: loopchain/channel/channel_service.py (hunk 2 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: I changed the SUBSCRIBE Call location inside a peer_manager

Comment (yakkle) [inline loopchain/channel/channel_service.py]: why don't you using with `enumerate` as following?

```suggestion
        for order, rep_info in enumerate(response["result"]["preps"], 1):
```

File: loopchain/channel/channel_service.py [hunk 5]
```diff
@@ -434,11 +432,46 @@ def __init_node_subscriber(self):
...
+        }
+
+        request = convert_params(request, ParamType.call)
+        stub = StubCollection().icon_score_stubs[ChannelProperty().name]
+        response = stub.sync_task().call(request)
+        response_to_json_query(response)
+
+        utils.logger.spam(f"from icon service channels is {response}")
+
+        peer_ids = ''
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]:
```python
from functools import reduce
from operator import add

peer_ids = (preps["id"] for preps in response["result"]["preps"])
peer_ids_appended = reduce(add, peer_ids)
```

File: loopchain/baseservice/peer_manager.py [hunk 6]
```diff
@@ -126,6 +126,34 @@ def __init__(self, channel_name):
...
+        :return:
+        """
+        group_id = conf.ALL_GROUP_ID
+
+        order_list = list(self.peer_order_list[group_id].keys())
+        order_list.sort()
+        peer_count = len(order_list)
+        peer_ids = ""
+
+        for i in range(peer_count):
```
Comment (zsaladin) [inline loopchain/baseservice/peer_manager.py]:
```python
from functools import reduce
from operator import add

peer_orders = self.peer_order_list[group_id]
peer_ids = (peer_orders[peer_order] for peer_order in sorted(peer_orders.keys()))
peer_ids_appended = reduce(add, peer_ids)
return self.get_peer_ids_hash(peer_ids_appended)
```

File: loopchain/baseservice/peer_manager.py (hunk 3 above)
Comment (windies21) [reply loopchain/baseservice/peer_manager.py]: It means list of rep (generator and validator)

**Review by zsaladin — APPROVED** [review]

**Review by laurenceyoon — APPROVED** [review]

**Pull request merged**